src/Microsoft.AspNet.FileProviders.Combined/CombinedFileProvider.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.FileProviders
+{
+    /// <summary>
+    /// Looks up files using embedded resources in the specified assembly.
+    /// This file provider is case sensitive.
+    /// </summary>
+    public class CombinedFileProvider : IFileProvider
+    {
+        private readonly IFileProvider[] _fileProviders;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinedFileProvider" /> class using a list of file provider.
+        /// </summary>
+        /// <param name="fileProviders"></param>
+        public CombinedFileProvider(params IFileProvider[] fileProviders)
```

Comment:
A `params` overload definitely makes sense, because that's how people will use this combined thingy: 

```csharp
options.FileProvider = new CombinedFileSystemProvider(
    new EmbeddedFileProvider(...),
    new EmbeddedFileProvider(...));
```

Since the file provider is a hot path, using `IEnumerable<IFileProvider>` is not really appropriate. Not to mention that, semantically, it's not immediately clear that the order matters when using `IEnumerable`. IMHO, an array or a list is much more appropriate in this case. That said, nothing prevents you from adding an `IEnumerable<>` overload that calls `ToArray()` or `ToList()` to avoid allocating enumerators in `GetFileInfo`/`GetDirectoryContents`.